fuse/readonly/readonly_unix.go

```diff
@@ -84,25 +94,37 @@ func (*Root) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
 // Node is the core object representing a filesystem tree node.
 type Node struct {
 	Ipfs   *core.IpfsNode
-	Nd     *mdag.ProtoNode
+	Nd     format.Node
 	cached *ftpb.Data
 }
 
 func (s *Node) loadData() error {
-	s.cached = new(ftpb.Data)
-	return proto.Unmarshal(s.Nd.Data(), s.cached)
+	if pbnd, ok := s.Nd.(*mdag.ProtoNode); ok {
+		s.cached = new(ftpb.Data)
+		return proto.Unmarshal(pbnd.Data(), s.cached)
+	}
+	return nil
 }
 
 // Attr returns the attributes of a given node.
 func (s *Node) Attr(ctx context.Context, a *fuse.Attr) error {
 	log.Debug("Node attr")
+	if rawnd, ok := s.Nd.(*mdag.RawNode); ok {
+		a.Mode = 0444
+		a.Size = uint64(len(rawnd.RawData()))
+		a.Blocks = 1
+		a.Uid = uint32(os.Getuid()) // TODO: should probably cache these calls. No sense making multiple syscalls for each attr call here
```

Comment:
Can we not just claim that these files are owned by root (does fuse not let us do that)?

Reply:
We can, I guess that does work.